File: packages/telemetry/typescript/examples/test_openai_agents.ts
/**
 * OpenAI Agents SDK — Latitude telemetry example.
 *
 * Required env vars:
 * - LATITUDE_API_KEY
 * - LATITUDE_PROJECT_SLUG
 * - OPENAI_API_KEY
 *
 * Install: npm install @openai/agents zod
 */

import { randomUUID } from "node:crypto"
import * as OpenAIAgentsSDK from "@openai/agents"
import { Agent, run, tool } from "@openai/agents"
import { z } from "zod"
import { capture, Latitude } from "../src"

const latitude = new Latitude({
  apiKey: process.env.LATITUDE_API_KEY!,
  project: process.env.LATITUDE_PROJECT_SLUG!,
  disableBatch: true,
  instrumentations: { "openai-agents": OpenAIAgentsSDK },
})

const PROVIDER = "openai-agents"
const MODEL = "gpt-5.5"
const SYSTEM = "You are a helpful assistant participating in a telemetry QA test. Keep answers concise."
const SESSION_ID = `${PROVIDER}-${randomUUID().slice(0, 8)}`

function ctx(scenario: string, ...extraTags: string[]) {
  return {
    tags: ["example", PROVIDER, "openai-agents-ts", ...extraTags],
    sessionId: SESSION_ID,
    userId: "example-user",
    metadata: { scenario, environment: "local" },
  }
}

const getWeather = tool({
  name: "get_weather",
  description: "Get the current weather for a city.",
  parameters: z.object({ city: z.string() }),
  execute: async ({ city }) => `The weather in ${city} is sunny and 21°C.`,
})

async function chat() {
  const agent = new Agent({ name: "Greeter", instructions: SYSTEM, model: MODEL })
  const result = await run(agent, "Say 'Hello from OpenAI Agents!' in exactly 5 words.")
  return result.finalOutput
}

async function toolConversation() {
  const agent = new Agent({
    name: "Weather agent",
    instructions: `${SYSTEM} Always call get_weather first.`,
    tools: [getWeather],
    model: MODEL,
  })
  const result = await run(
    agent,
    "What's the weather in San Francisco? Use get_weather, then answer in one short sentence.",
  )
  return result.finalOutput
}

async function main() {
  await latitude.ready

  await capture("openai-agents-chat-capture", chat, ctx("chat"))
  await capture("openai-agents-tools-capture", toolConversation, ctx("tools", "tools"))

  await latitude.flush()
  await latitude.shutdown()
}

main().catch((err) => {
  console.error(err)
  process.exit(1)
})

File: packages/telemetry/python/examples/test_openai_agents.py
"""
OpenAI Agents SDK — Latitude telemetry example.

Required env vars:
- LATITUDE_API_KEY
- LATITUDE_PROJECT_SLUG
- OPENAI_API_KEY

Install: uv add openai-agents
"""

import asyncio
import os
import uuid

import agents
from agents import Agent, Runner, function_tool

from latitude_telemetry import Latitude, capture

latitude = Latitude(
    api_key=os.environ["LATITUDE_API_KEY"],
    project=os.environ["LATITUDE_PROJECT_SLUG"],
    instrumentations={"openai-agents": agents},
    disable_batch=True,
)

PROVIDER = "openai-agents"
MODEL = "gpt-4o-mini"
SESSION_ID = f"{PROVIDER}-{uuid.uuid4().hex[:8]}"


def _ctx(scenario: str, *extra_tags: str) -> dict:
    return {
        "tags": ["example", PROVIDER, *extra_tags],
        "session_id": SESSION_ID,
        "user_id": "example-user",
        "metadata": {"scenario": scenario, "environment": "local"},
    }


@function_tool
def get_weather(city: str) -> str:
    """Get the current weather for a city."""
    return f"The weather in {city} is sunny and 21°C."


def chat() -> str:
    agent = Agent(name="Greeter", instructions="Answer concisely.", model=MODEL)
    return asyncio.run(Runner.run(agent, "Say 'Hello from OpenAI Agents!' in exactly 5 words.")).final_output


def tool_conversation() -> str:
    agent = Agent(
        name="Weather agent",
        instructions="Answer weather questions concisely. Always call get_weather first.",
        tools=[get_weather],
        model=MODEL,
    )
    result = asyncio.run(
        Runner.run(agent, "What's the weather in San Francisco? Use get_weather, then answer in one short sentence.")
    )
    return result.final_output


if __name__ == "__main__":
    tool_conversation()

    capture("openai-agents-chat-capture", chat, _ctx("chat"))
    capture("openai-agents-tools-capture", tool_conversation, _ctx("tools", "tools"))

    latitude.flush()
